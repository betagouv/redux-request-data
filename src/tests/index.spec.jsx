import 'babel-polyfill'

import { mount, shallow } from 'enzyme'
import React from 'react'
import { Provider } from 'react-redux'

import { configureTestStore,
  configureFetchDataWithRequestFail,
  configureFetchDataWithRequestSuccess
} from './configure'
import Foos, { RawFoos } from './Foos'

describe('src | components | pages | hocs | withRequest', () => {

  beforeEach(() => {
    fetch.resetMocks()
  })

  describe('snapshot', () => {
    it('should match snapshot', () => {
      // when
      const wrapper = shallow(<RawFoos dispatch={jest.fn()} foos={[]} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mount with request', () => {
    describe('request with success', () => {
      it('should render test component whith foo items', done => {
        // when
        const store = configureTestStore()
        configureFetchDataWithRequestSuccess()

        // then
        mount(
          <Provider store={store}>
            <Foos onSuccessUpdateCallback={done} />
          </Provider>
        )
      })
    })

    describe('request with fail', () => {
      it('should render test component whith no foo items', done => {
        // when
        const store = configureTestStore()
        configureFetchDataWithRequestFail()

        // then
        mount(
          <Provider store={store}>
            <Foos onFailUpdateCallback={done} />
          </Provider>
        )
      })
    })
  })
})
