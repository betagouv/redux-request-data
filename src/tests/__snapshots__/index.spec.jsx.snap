// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | hocs | withRequest snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RawFoos
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "config": Object {
                "apiPath": "/foos",
                "handleFail": [Function],
                "method": "GET",
              },
              "type": "REQUEST_DATA_GET_/FOOS",
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    foos={Array []}
    onFailUpdateCallback={[Function]}
    onSuccessUpdateCallback={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
